include/pagmo/algorithms/gaco.hpp

```diff
@@ -0,0 +1,988 @@
+                if (m_memory == true) {
+                    m_sol_archive = sol_archive;
+                }
+            }
+
+            // 3 - Logs and prints (verbosity modes > 1: a line is added every m_verbosity generations)
+
+            if ((m_verbosity > 0u && gen != m_gen) || (m_verbosity > 0u && m_memory == true)) {
+                // Every m_verbosity generations print a log line
+                if (gen % m_verbosity == 1u || m_verbosity == 1u) {
+                    auto best_fit = sol_archive[0][1 + n_x];
+                    double dx = 0., dp = 0.;
+                    // The population flattness in variables
+                    for (decltype(n_x) i = 0u; i < n_x; ++i) {
+                        dx += std::abs(sol_archive[m_ker - 1][1 + i] - sol_archive[0][1 + i]);
+                    }
+                    // The population flattness in penalty
+                    dp = std::abs(sol_archive[m_ker - 1][0] - sol_archive[0][0]);
+                    // Every line print the column names
+                    if (m_memory == false) {
+                        if (count_screen % 50u == 1u) {
+                            print("\n", std::setw(7), "Gen:", std::setw(15), "Fevals:", std::setw(15),
+                                  "Best:", std::setw(15), "Kernel:", std::setw(15), "Oracle:", std::setw(15),
+                                  "dx:", std::setw(15), std::setw(15), "dp:", '\n');
+                        }
+
+                    } else if ((m_memory == true && m_counter == 1) || (m_memory == true && m_counter % 50u == 1u)) {
+                        print("\n", std::setw(7), "Gen:", std::setw(15), "Fevals:", std::setw(15),
+                              "Best:", std::setw(15), "Kernel:", std::setw(15), "Oracle:", std::setw(15),
+                              "dx:", std::setw(15), std::setw(15), "dp:", '\n');
+                    }
+                    print(std::setw(7), gen, std::setw(15), m_fevals, std::setw(15), best_fit, std::setw(15), m_ker,
+                          std::setw(15), m_oracle, std::setw(15), dx, std::setw(15), dp, '\n');
+
+                    ++count_screen;
+                    // Logs
+                    m_log.emplace_back(gen, m_fevals, best_fit, m_ker, m_oracle, dx, dp);
+                }
+            }
+
+            // 4 - compute pheromone values
+
+            vector_double sigma(n_x);
+            pheromone_computation(gen, prob_cumulative, omega, sigma, pop, sol_archive);
+
+            // 5 - use pheromone values to generate new ants (i.e., individuals)
+
+            // I create the vector of vectors where I will store all the new ants which will be generated:
+            std::vector<vector_double> new_ants(pop_size, vector_double(n_x, 1));
+            generate_new_ants(popold, dist, gauss, prob_cumulative, sigma, new_ants, sol_archive);
+
+            for (population::size_type i = 0; i < pop_size; ++i) {
+                vector_double ant(n_x);
```

Comment:
Same here. If possible and if it does not hurt the readability of the code, it would be better to have some temp vectors declared outside of the loop and re-used in these inner loops.